src/test/java/org/elasticsearch/action/IndicesRequestTests.java

```diff
@@ -480,6 +511,12 @@ public void testIndicesStats() {
 
         clearInterceptedActions();
         assertSameIndices(indicesStatsRequest, indicesStats);
+
+        indicesStatsRequest = new IndicesStatsRequest().indices("v*");
+        IndicesStatsResponse response = internalCluster().clientNodeClient().admin().indices().stats(indicesStatsRequest).actionGet();
+
+        clearInterceptedActions();
+        assertThat("'v*' should not match any indices, so no shard should be returned", response.getTotalShards(), equalTo(0));
```

Comment:
same as above, also I wonder if we could unit test `RoutingTable` instead of needing an integration tests to verify how many shards are touched...